### DISCUSSION TITLE
Conversations are being forked without the user requesting a fork

### DISCUSSION THREAD

**Discussion by malagna-amplify** [danny-avila/LibreChat#3381] [Troubleshooting] [answered]
### What happened?

If I switch from one conversation back to another conversation, and type a message into the chat, sometimes that message is treated as a fork of the original question in the conversation, instead of a new message in the conversation. 

### Steps to Reproduce

1. Create one conversation
2. Create another conversations
3. Switch back to first conversation & type something into the chat
4. Message shows up as a fork of the original message rather than a continuation of the existing conversation. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
This can't be recreated, can you confirm what version of LibreChat you are using?

**Reply by malagna-amplify** [reply]
Here are the screenshots if me reproducing the error - maybe I am not using the right terminology. Issue happens after switching between chats multiple times. Using librechat 0.7.4

<img width="986" alt="Screen Shot 2024-07-18 at 2 22 03 PM" src="https://github.com/user-attachments/assets/0d11dd9c-9a25-44f9-b148-6e8f8e7f0c10">
<img width="983" alt="Screen Shot 2024-07-18 at 2 22 32 PM" src="https://github.com/user-attachments/assets/28462eec-8717-4c88-9d97-058c34f63a2b">
<img width="986" alt="Screen Shot 2024-07-18 at 2 22 46 PM" src="https://github.com/user-attachments/assets/4f469322-4b59-45c4-be33-d59686e46b44">
<img width="989" alt="Screen Shot 2024-07-18 at 2 22 55 PM" src="https://github.com/user-attachments/assets/de5dc049-5b42-45bd-847d-af344edef452">
<img width="997" alt="Screen Shot 2024-07-18 at 2 23 08 PM" src="https://github.com/user-attachments/assets/f34250cd-a09d-49fd-a1ba-66d4989d5359">
<img width="1002" alt="Screen Shot 2024-07-18 at 2 23 21 PM" src="https://github.com/user-attachments/assets/ec9b0b5a-9b9f-45d0-8964-c62f61ac9bd8">
<img width="993" alt="Screen Shot 2024-07-18 at 2 23 30 PM" src="https://github.com/user-attachments/assets/fc2f07f6-f056-490e-8467-fd799d2608bb">

**Reply by malagna-amplify** [reply]
In this example the question "Where is italy?" replaced the original question as option 2/2 instead of being added to the existing conversation.

**Reply by malagna-amplify** [reply]
It looks like something strange is happening with the parentMessageId and responseMessageId. Also worth noting are using the custom api, since our requests to LLMs go through a reverse proxy. 

<img width="353" alt="Screen Shot 2024-07-19 at 12 52 10 PM" src="https://github.com/user-attachments/assets/caf95e36-e760-4427-91ca-ddf0b5ba0027">

**Reply by malagna-amplify** [reply]
I just reproduced the issue with a fresh clone of the librechat repo, using the standard openAI provider. I think it has something to do with the frontend context management. Maybe one of the hooks not updating properly when switching back and forth between conversations multiple times.

**Reply by danny-avila** [reply]
thanks @malagna-amplify I'm looking into it

**Reply by danny-avila** [reply]
@malagna-amplify I think this is now taken care of, can you confirm I recreated this well?

https://github.com/user-attachments/assets/7a34cd04-966b-4792-8cb8-0e2dd049c396

**Reply by danny-avila** [reply]
By now I mean I will push an update within the hour

**Reply by malagna-amplify** [reply]
No I don't think your recording captures what I am talking about. Here, I have attached a recording of the bug being reproduced. 

https://github.com/user-attachments/assets/4d9c0618-86ce-4795-92cb-0cdb2c58d011

**Reply by malagna-amplify** [reply, marked as answer]
@danny-avila  I confirmed that with your latest changes, the issue appears to be resolved. Thank you!
